pkg/proc/bininfo.go

```diff
@@ -61,9 +61,12 @@ type BinaryInfo struct {
 	packageMap map[string]string
 
 	frameEntries frame.FrameDescriptionEntries
-	compileUnits []*compileUnit
-	types        map[string]dwarfRef
-	packageVars  []packageVar // packageVars is a list of all global/package variables in debug_info, sorted by address
+
+	compileUnits         []*compileUnit
+	compileUnitsByOffset []*compileUnit
```

Comment:
How often will this actually be used in practice? I'm considering the cost of increased default memory usage storing each pointer twice vs just sorting on demand.

Reply:
AFAICT it's only used for a workaround that should never happen with never version of Go, so it should be used almost never. It occupies 8*num_packages so to get to ~1MB the project would have to use 100000 packages, which is a lot (delve has 160).

Reply:
Right, it seems like this is a convenience method, it's not used anywhere else except the test in this CL. I think for that reason it's better to pay the cost of a slower search vs (even minimal) increased memory usage.

Reply:
Done.

Reply:
Is there any reason not to always sort compile units by offset anyways?

Reply:
Apparently, no. We can't use the sort-by-pc we're doing, probably because they are not contiguous.

Reply:
What are your thoughts on that then, just sort that list by offset always, and then reimplement the sort.Search you had originally which was a bit cleaner.

Reply:
Done